edx_event_bus_kafka/internal/producer.py

```diff
@@ -157,6 +167,39 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
     return key_serializer, value_serializer
 
 
+class ProducingContext:
+    """
+    Wrapper class to allow us to link a call to produce() with the on_event_deliver callback
+    """
+    def __init__(self, context_data):
+        self.context_data = context_data
+
+    def __str__(self):
+        return " ".join([f"{key}={value}" for key, value in self.context_data.items()])
+
+    def set_context_data(self, key, value):
+        self.context_data[key] = value
```

Comment:
Rather than a dictionary, can ProducingContext just be an attrs class or dataclass with these keys as attributes? I think it might save a bunch of code. Wouldn't need as many string constants either.

Reply:
[fixed]